src/panels/todo/dialog-todo-item-editor.ts

```diff
@@ -0,0 +1,403 @@
+import { CSSResultGroup, LitElement, css, html, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-date-input";
+import "../../components/ha-textarea";
+import "../../components/ha-time-input";
+import {
+  TodoItemStatus,
+  createItem,
+  deleteItems,
+  updateItem,
+} from "../../data/todo";
+import { TimeZone } from "../../data/translation";
+import { showConfirmationDialog } from "../../dialogs/generic/show-dialog-box";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { TodoItemEditDialogParams } from "./show-dialog-todo-item-editor";
+
+@customElement("dialog-todo-item-editor")
+class DialogTodoItemEditor extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _error?: string;
+
+  @state() private _params?: TodoItemEditDialogParams;
+
+  @state() private _summary = "";
+
+  @state() private _description? = "";
+
+  @state() private _due?: Date;
+
+  @state() private _checked = false;
+
+  @state() private _hasTime = false;
+
+  @state() private _submitting = false;
+
+  // Dates are manipulated and displayed in the browser timezone
+  // which may be different from the Home Assistant timezone. When
+  // events are persisted, they are relative to the Home Assistant
+  // timezone, but floating without a timezone.
+  private _timeZone?: string;
+
+  public showDialog(params: TodoItemEditDialogParams): void {
+    this._error = undefined;
+    this._params = params;
+    this._timeZone =
+      this.hass.locale.time_zone === TimeZone.local
+        ? Intl.DateTimeFormat().resolvedOptions().timeZone
+        : this.hass.config.time_zone;
+    if (params.item) {
+      const entry = params.item;
+      this._checked = entry.status === TodoItemStatus.Completed;
+      this._summary = entry.summary;
+      this._description = entry.description || "";
+      this._due = entry.due ? new Date(entry.due) : undefined;
+      this._hasTime = entry.due?.includes("T") || false;
+    } else {
+      this._hasTime = false;
+      this._checked = false;
+      this._due = undefined;
+    }
+  }
+
+  public closeDialog(): void {
+    if (!this._params) {
+      return;
+    }
+    this._error = undefined;
+    this._params = undefined;
+    this._due = undefined;
+    this._summary = "";
+    this._description = "";
+    this._hasTime = false;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._params) {
+      return nothing;
+    }
+    const isCreate = this._params.item === undefined;
+
+    const { dueDate, dueTime } = this._getLocaleStrings(this._due);
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this.closeDialog}
+        scrimClickAction
+        escapeKeyAction
```

Comment:
```suggestion
```
Do you think we can allow escape to easily close the popup if the user misclick the checkbox and it opens the dialog?

Reply:
The rule is to not allow it on dialogs with input forms 🤷